### PR TITLE
Classifier auth

### LINKED ISSUE #515: Add auth headers to classifier API calls
**Package**
lib-classifier

**Feature or Issue Description**
The classifier request subjects and submits classifications as an anonymous user. Authorisation headers should be added to the API calls.

### PR DESCRIPTION
**Pull request by eatyourgreens**
Package:
lib-classifier

Closes #515.

Describe your changes:
Adds Authorization headers to subject requests and classification posts.

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?
- [ ] Is the changelog updated?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?

### CODE REVIEW

File: packages/lib-classifier/src/store/utils/ClassificationQueue.js [hunk 1]
```diff
@@ -3,15 +3,17 @@
...
 const FAILED_CLASSIFICATION_QUEUE_NAME = 'failed-classifications'
 const MAX_RECENTS = 10
 const RETRY_INTERVAL = 5 * 60 * 1000
 
 class ClassificationQueue {
-  constructor (api, onClassificationSaved) {
+  constructor (api, onClassificationSaved, authClient) {
     this.storage = window.localStorage
     this.apiClient = api || panoptes
+    this.authClient = authClient
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/store/utils/ClassificationQueue.js]: Do we want a default for the auth client here? If so, what should it be?

File: packages/lib-classifier/src/store/utils/ClassificationQueue.js [hunk 2]
```diff
@@ -58,33 +60,36 @@ class ClassificationQueue {
...
-
-          if (error.status !== 422) {
-            try {
-              this.store(classificationData)
-              if (!this.flushTimeout) {
-                this.flushTimeout = setTimeout(this.flushToBackend.bind(this), RETRY_INTERVAL)
+    return getBearerToken(this.authClient)
+      .then((authorization) => {
+        return Promise.all(pendingClassifications.map((classificationData) => {
+          return this.apiClient.post(this.endpoint, { classifications: classificationData }, authorization)
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/store/utils/ClassificationQueue.js]: The nested Promises here are becoming hard to follow.

Comment (srallen) [reply packages/lib-classifier/src/store/utils/ClassificationQueue.js]: This could be refactored to use async/await

File: packages/lib-classifier/src/store/utils/ClassificationQueue.js (hunk 1 above)
Comment (rogerhutchings) [reply packages/lib-classifier/src/store/utils/ClassificationQueue.js]: I don't think we do - we should add a console statement saying there is no auth client, but we shouldn't require one / default to one.

Comment (eatyourgreens) [reply packages/lib-classifier/src/store/utils/ClassificationQueue.js]: Authorization headers aren't required on classifications, so I'm happy to default this to `undefined`.

File: packages/lib-classifier/src/store/utils/ClassificationQueue.js (hunk 2 above)
Comment (rogerhutchings) [reply packages/lib-classifier/src/store/utils/ClassificationQueue.js]: cc #525
